keysign/network/AvahiBrowser.py

```diff
@@ -18,50 +18,28 @@
 #    You should have received a copy of the GNU General Public License
 #    along with GNOME Keysign.  If not, see <http://www.gnu.org/licenses/>.
 from __future__ import print_function
-import avahi, dbus
-from dbus import DBusException
-from dbus.mainloop.glib import DBusGMainLoop
+import logging
+import os
 
-from gi.repository import Gio
+import dbus
+from dbus.mainloop.glib import DBusGMainLoop
 from gi.repository import GObject
 
-import logging
+if __name__ == "__main__" and __package__ is None:
+    logging.getLogger().error("You seem to be trying to execute " +
+                              "this script directly which is discouraged. " +
+                              "Try python -m instead.")
+    parent_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    os.sys.path.insert(0, parent_dir)
+    os.sys.path.insert(0, os.path.join(parent_dir, 'monkeysign'))
+    __package__ = str('keysign')
+
+from .AvahiConstants import AvahiConstants as avahi
 
 __all__ = ["AvahiBrowser"]
 
 DBusGMainLoop( set_as_default=True )
 
-# This should probably be upstreamed.
-# Unfortunately, upstream seems rather inactive.
-if getattr(avahi, 'txt_array_to_dict', None) is None:
-    # This has been taken from Gajim
-    # https://dev.gajim.org/gajim/gajim/blob/master/src/common/zeroconf/zeroconf_avahi.py
-    # it is licensed under the GPLv3.
-    # https://github.com/lathiat/avahi/pull/133
-    def txt_array_to_dict(txt_array):
```

Comment:
I don't think it's clever to remove, because the upstream avahi doesn't ship this function.